lib/spack/docs/build_settings.rst

```diff
@@ -209,6 +209,170 @@ Specific limitations include:
+
+.. code-block:: console
+
+    $ module purge
+    $ module load gcc/8.3.1
+    $ module load cuda/10.1.2
+    $ module show openmpi/4.0.2-cuda
+    -------------------------------------------------------------------
+    /gpfslocalsup/pub/modules-idris-env4/modulefiles/linux-rhel8-skylake_avx512/openmpi/4.0.2-cuda:
+
+    module-whatis   {An open source Message Passing Interface implementation.}
+    prereq          intel-compilers/19.0.4 pgi/20.1 pgi/19.10 gcc/10.1.0 gcc/8.3.1
+    prereq          cuda/10.2 cuda/10.1.2 cuda/10.1.1
+    conflict        openmpi
+    conflict        intel-mpi
+    prepend-path    CPATH /gpfslocalsup/spack_soft/openmpi/4.0.2/gcc-8.3.1-n6vcsair26tkpepojy3c2gqxtqccijq3/include
+    prepend-path    LD_LIBRARY_PATH /gpfslocalsup/spack_soft/openmpi/4.0.2/gcc-8.3.1-n6vcsair26tkpepojy3c2gqxtqccijq3/lib
+    prepend-path    LIBRARY_PATH /gpfslocalsup/spack_soft/openmpi/4.0.2/gcc-8.3.1-n6vcsair26tkpepojy3c2gqxtqccijq3/lib
+    prepend-path    PATH /gpfslocalsup/spack_soft/openmpi/4.0.2/gcc-8.3.1-n6vcsair26tkpepojy3c2gqxtqccijq3/bin
+    [snip]
+
+This output can be parsed by Spack when building software. To obtain an entry
+for this external package in the ``package.yaml`` file, you could run ``spack
+external find openmpi`` after loading the dependencies to benefit from the
+automatic variant detection of the OpenMPI build. Afterwards, add the module
+and all of its dependencies in the ``packages.yaml`` file to arrive at the
+following OpenMPI entry:
+
+.. code-block:: yaml
+
+    packages:
+      openmpi:
+        externals:
+        - spec: openmpi@4.0.2+cuda+cxx+cxx_exceptions~java~memchecker+pmi~sqlite3+static~thread_multiple~wrapper-rpath
+            fabrics=psm2 schedulers=slurm
+          prefix: /gpfslocalsup/spack_soft/openmpi/4.0.2/gcc-8.3.1-n6vcsair26tkpepojy3c2gqxtqccijq3
+          modules: [gcc/8.3.1, cuda-10.1.2, openmpi/4.0.2-cuda]
+
+Once all dependencies are satisfied or if there are no dependencies, then the
+prefix determined by Spack may not be correct in the presence of nonstandard
+paths. This is rarely the case because Spack contains package-specific code to
+deal with these quirks. Before manually setting the prefix in the
+``packages.yaml`` file in addition to a list of modules, it is strongly
+suggested to check the package name and the module list again. Consider the
+OpenMPI module on CentOS 7:
+
+.. code-block:: console
+
+    [john@c7 ~]# module show mpi
+    -------------------------------------------------------------------
+    /etc/modulefiles/mpi/openmpi-x86_64:
+
+    conflict	 mpi
+    prepend-path	 PATH /usr/lib64/openmpi/bin
+    prepend-path	 LD_LIBRARY_PATH /usr/lib64/openmpi/lib
+    prepend-path	 PYTHONPATH /usr/lib64/python2.7/site-packages/openmpi
+    prepend-path	 MANPATH /usr/share/man/openmpi-x86_64
+    [snip]
+
+Given this module output, ``spack external find`` will determine
+``/usr/lib64/openmpi`` as the prefix when it is actually ``/usr``. This
```

Comment:
Are there also headers in this case? Where are they stored? If it's something like `/usr/include/openmpi` then FWIW there is no prefix that Spack can use generally: we expect that all the package's files are contained in a single prefix for now, e.g. `/usr/{lib, include}` or `/path/to/openmpi/{lib, include}`. There might be per-package support for this in a couple cases, but generally an external which behaves this way will require symlinked directories to create the illusion of a unified prefix.